File: examples/objective-c-demo/objective-c-demo/HomeNavViewController.m
 
#import "HomeNavViewController.h"
#import "PlayerViewController.h"
#import "AliPlayerViewController.h"

@implementation HomeNavViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
}

- (void)goAliPlayerVideoBtnAction {
    AliPlayerViewController *vc = [[AliPlayerViewController alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)goPlayerVideoBtnAction {
    PlayerViewController *vc = [[PlayerViewController alloc] init];
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)setupUI {
    self.view.backgroundColor = [UIColor blackColor];
    CGFloat btnHeight = 50.0;
    
    self.aliPlayerButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 80, self.view.frame.size.width, btnHeight)];
    [self.aliPlayerButton setTitle:@"AliPlayer" forState:UIControlStateNormal];
    [self.aliPlayerButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    
    self.playerButton = [[UIButton alloc] initWithFrame:CGRectMake(0, CGRectGetMaxY(self.aliPlayerButton.frame), self.view.frame.size.width, btnHeight)];
    [self.playerButton setTitle:@"AVPlayer" forState:UIControlStateNormal];
    [self.playerButton setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    
    [self.aliPlayerButton addTarget:self action:@selector(goAliPlayerVideoBtnAction) forControlEvents:UIControlEventTouchUpInside];
    [self.playerButton addTarget:self action:@selector(goPlayerVideoBtnAction) forControlEvents:UIControlEventTouchUpInside];
    
    [self.view addSubview:self.aliPlayerButton];
    [self.view addSubview:self.playerButton];
}
@end

File: examples/swift-demo/swift-demo/HomeNavViewController.swift

import MLYSDK
import UIKit

let clientID = "cegh8d9j11u91ba1u600"
let videoURL = "https://vsp-stream.s3.ap-northeast-1.amazonaws.com/HLS/raw/SpaceX.m3u8" 

class HomeNavViewController: UIViewController {
     
    var aliPlayerButton: UIButton!
    
    var playerButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        self.setupUI()
    }
 
    @objc func goAliPlayerVideoBtnAction() {
        self.navigationController?.pushViewController(AliPlayerViewController(), animated: true)
    }
    
    @objc func goPlayerVideoBtnAction() {
        self.navigationController?.pushViewController(PlayerViewController(), animated: true)
    }
 
}

extension HomeNavViewController {
    func setupUI() {
        self.view.backgroundColor = .black
        let btnHeight = 50.0
   
        self.aliPlayerButton = UIButton(frame: CGRect(x: 0, y: 80, width: self.view.frame.width, height: btnHeight))
        self.aliPlayerButton.setTitle("AliPlayer", for: .normal)
        self.aliPlayerButton.setTitleColor(.white, for: .normal)
        
        self.playerButton = UIButton(frame: CGRect(x: 0, y: self.aliPlayerButton.frame.maxY, width: self.view.frame.width, height: btnHeight))
        self.playerButton.setTitle("AVPlayer", for: .normal)
        self.playerButton.setTitleColor(.white, for: .normal)
 
        self.aliPlayerButton.addTarget(self, action: #selector(self.goAliPlayerVideoBtnAction), for: .touchUpInside)
        self.playerButton.addTarget(self, action: #selector(self.goPlayerVideoBtnAction), for: .touchUpInside)
 
        self.view.addSubview(self.aliPlayerButton)
        self.view.addSubview(self.playerButton)
 
    }
}
